src/elements/EditableTable.tsx

```diff
@@ -0,0 +1,353 @@
+    const footerRow: FooterRow = calculateFooterRow(cols, rows);
+
+    const handleMouseEnter = (event: React.MouseEvent, row: RowData) => {
+        raiseEvent("custom", { eventType: "mouseEnter", row });
+        onMouseEnter?.(row);
+    };
+
+    const handleMouseLeave = (event: React.MouseEvent, row: RowData) => {
+        raiseEvent("custom", { eventType: "mouseLeave", row });
+        onMouseLeave?.(row);
+    };
+
+    const handleCellChange = (value: string | number, row: RowData, column: Column) => {
+        if (column.type === "number") {
+            row[column.name] = Number(value);
+        } else {
+            row[column.name] = value;
+        }
+
+        if (rowCalculation) {
+            rowCalculation(row[column.name], row, column);
+        }
+        setEditingRows({ ...editingRows });
+    };
+
+    const handleToggleEdit = (event: React.MouseEvent, index: number, rows: RowData[]) => {
+        const row = rows[index];
+        const editRow = editingRows[index];
+
+        if (editRow) {
+            rows[index] = editingRows[index];
+            delete editingRows[index];
+        } else {
+            editingRows[index] = JSON.parse(JSON.stringify(row));
+        }
+        setData([...rows]);
+    };
+
+    const handleCancel = (event: React.MouseEvent, index: number, rows: RowData[]) => {
+        setData([...rows]);
+        delete editingRows[index];
+    };
+    return (
+        <Box maxHeight={maxHeight} maxWidth={maxWidth} sx={{ overflow: "auto" }}>
+            <Table
+                size={size}
+                stickyHeader={stickyHeader}
+                sx={{
+                    // Hack required to override .gcx-forms.defaults
+                    ["& span"]: {
+                        display: "inline-flex !important",
+                    },
+                }}
+            >
+                <TableHead>
+                    <TableRow>
+                        <TableCell />
+                        {cols.map((col) => (
+                            <TableCell align={col.align} key={col.name}>
+                                {col.alias ?? col.name}
+                            </TableCell>
+                        ))}
+                    </TableRow>
+                </TableHead>
+                <TableBody>
+                    {data.map((row, index) => {
+                        return (
+                            <TableRow
+                                hover
+                                key={index}
+                                onMouseEnter={(event) => handleMouseEnter(event, row)}
+                                onMouseLeave={(event) => handleMouseLeave(event, row)}
+                            >
+                                <TableCell sx={{ ...TableCellStyleOverrides }} key={index}>
+                                    {editingRows && editingRows[index] ? (
+                                        <Box sx={{ ".MuiBox-root": { width: "fit-content" } }}>
```

Comment:
I don't think this `sx` is actually getting applied. I replaced the `<Box>` with `<>` and it looked the same.